src/main/java/com/exasol/adapter/dynamodb/documentfetcher/dynamodb/DynamodbScanDocumentFetcherFactory.java

```diff
@@ -0,0 +1,54 @@
+package com.exasol.adapter.dynamodb.documentfetcher.dynamodb;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.amazonaws.services.dynamodbv2.model.ScanRequest;
+import com.exasol.adapter.dynamodb.documentfetcher.DocumentFetcher;
+import com.exasol.adapter.dynamodb.documentnode.dynamodb.DynamodbNodeVisitor;
+import com.exasol.adapter.dynamodb.remotetablequery.RemoteTableQuery;
+
+/**
+ * This factory builds {@link DynamodbScanDocumentFetcher}s for a given query. It returns multiple
+ * {@link DocumentFetcher}s so that their execution can be parallelized.
+ */
+class DynamodbScanDocumentFetcherFactory {
+
+    /**
+     * Build {@link DynamodbScanDocumentFetcher}s for a given query.
+     * 
+     * @param remoteTableQuery query to build the {@link DynamodbScanDocumentFetcher}s for
+     * @return list of {@link DynamodbScanDocumentFetcher}s
+     */
+    public List<DocumentFetcher<DynamodbNodeVisitor>> buildDocumentFetcherForQuery(
+            final RemoteTableQuery<DynamodbNodeVisitor> remoteTableQuery) {
+        final ScanRequest templateScanRequest = new ScanRequest()
+                .withTableName(remoteTableQuery.getFromTable().getRemoteName());
+        final DynamodbAttributeNamePlaceholderMapBuilder namePlaceholderMapBuilder = new DynamodbAttributeNamePlaceholderMapBuilder();
+        final DynamodbAttributeValuePlaceholderMapBuilder valuePlaceholderMapBuilder = new DynamodbAttributeValuePlaceholderMapBuilder();
+        final String filterExpression = new DynamodbFilterExpressionFactory(namePlaceholderMapBuilder,
+                valuePlaceholderMapBuilder).buildFilterExpression(remoteTableQuery.getSelection());
+        if (!namePlaceholderMapBuilder.getPlaceholderMap().isEmpty()) {
+            templateScanRequest.setExpressionAttributeNames(namePlaceholderMapBuilder.getPlaceholderMap());
+        }
+        if (!filterExpression.isEmpty()) {
+            templateScanRequest.setFilterExpression(filterExpression);
+        }
+        if (!valuePlaceholderMapBuilder.getPlaceholderMap().isEmpty()) {
+            templateScanRequest.setExpressionAttributeValues(valuePlaceholderMapBuilder.getPlaceholderMap());
+        }
+        return parallelizeScan(templateScanRequest);
+    }
+
+    private List<DocumentFetcher<DynamodbNodeVisitor>> parallelizeScan(final ScanRequest templateScanRequest) {
+        final int numberOfSegments = 10;
```

Comment:
Is there a specific reason for this number 10?

Reply:
No. Right now it's just hard coded so there is some paralelization. In the future the number of segments should be generated automatically. I added a todo note.